skimage/feature/corner.py

```diff
@@ -950,13 +950,20 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
     array([[2, 2]])
 
     """
+    if threshold_rel is None:
+        threshold_rel = 0.1
+        warn("Until the version 0.16, threshold_rel was set to 0.1 by default."
+             "Starting from version 0.16, the default value is set to None."
+             "Until version 0.18, a None value corresponds to a threshold value of 0.1."
+             "The default behavior will match skimage.feature.peak_local_max.")
```

Comment:
I think this should say what the `peak_local_max` default actually is...

Reply:
It defaults to `max(threshold_abs, threshold_rel * image.max())` and `threshold_abs` defaults to `image.min()`. The connection between `threshold_abs` and `threshold_rel` is not documented in `peak_local_max`. Might be better to explain it there and just pointed the warning at that explanation considering that it's not trivial and this function's docstring already does the same.

Reply:
@sciunto could you add `stacklevel=2` and `category=FutureWarning` here?

Reply:
I'm OK with the idea to improve docstrings, but i would like to leave it for another PR.